src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -0,0 +1,55 @@
+package com.sourcegraph.cody.history
+
+import com.intellij.openapi.components.*
+import com.sourcegraph.cody.agent.protocol.ChatMessage
+import com.sourcegraph.cody.agent.protocol.Speaker
+import com.sourcegraph.cody.history.state.ChatState
+import com.sourcegraph.cody.history.state.HistoryState
+import com.sourcegraph.cody.history.state.MessageState
+import java.time.LocalDateTime
+
+@State(
+    name = "com.sourcegraph.cody.history.HistoryService", storages = [Storage("cody_history.xml")])
+@Service(Service.Level.PROJECT)
+class HistoryService : SimplePersistentStateComponent<HistoryState>(HistoryState()) {
+
+  private val listeners = mutableListOf<(ChatState) -> Unit>()
```

Comment:
Don't you need to `synchronize` operation on this list?

Reply:
Thanks, now added.